doc/integrator/ngeo.rst.mako

```diff
@@ -113,3 +113,112 @@ The dynamic values names are: ``interface``, ``cache_version``, ``lang_urls``, `
+
+Create your own component
+-------------------------
+
+Create your ``.js`` and ``.html`` (partial) files in a dedicated folder under:
+``geoportal/<package>_geoportal/static-ngeo/js/``. We encourage you to write standard ECMAScript and
+use AngularJS only to display your component.
+
+For the structure, you can be inspired by one of the components in ngeo.
+
+You should register each your AngularJS element (component, controller, service, etc.) in a module. First
+declare your module:
+
+.. code:: javascript
+
+   import angular from 'angular';
+   const module = angular.module('<projectElementname>', [
+     // dependencies to load within this module.
+   ]);
+
+Then register your component to this module:
+
+.. code:: javascript
+
+   class aController {}
+   module.controller('<projectController>', aController);
+   const aDirective = function() {};
+   module.directive('<projectDirectivename>', aDirective);
+
+And export this module:
+
+.. code:: javascript
+
+   export default module;
+
+Finally add your module in your main controller app file:
+
+.. code:: javascript
+
+   import myCustomComponent from '../myComponent/Component.js';
+   // Update the already declared module at the end of the controller with:
+   const module = angular.module('<AppInterface>', [
+     ...
+     myCustomComponent.name,
+   ];
+
+At this point, if in your <interface>.html app file you declare your component, it should be loaded:
+
+.. code:: html
+
+   <project-directivename></project-directivename>
+
+If you have no error and no result, verify that the module is correctly declared and loaded. AngularJS can't
+throw errors for a component that it doesn't know and having a custom markup in the html is not an error
+for the html.
+
+.. note::
+
+To load a partial (html) file for a component you can use this syntaxe inside the declaration of your
+component: ``template: require('./component.html')`` (from ``./Component.js`` file).
+
+Extend or override an existing component
+----------------------------------------
+
+Overriding an existing partial (html) or js element is possible but you have to test carefully your code
+after each GMF update as the original code may have changed.
+
+AngularJS is still ECMAScript. You can define some new functions or new variables to a class via the
+prototype. Here's an example with the ngeo draw controller:
+
+.. code:: javascript
+    
+   import `import {DrawController} from 'ngeo/draw/Controller.js'`
+   /**
+    * Doctype is important for compilation.
+    * @type {string}
+    */
+   DrawController.prototype.myNewValue;
+   DrawController.prototype.newFunction = function () {};
+
+To replace an existing function, just create a function with the same name that the one to replace:
+
+.. code:: javascript
+
+   DrawController.prototype.$onInit = function () {};
+
+You will have to copy-paste the content of the original function to keep the component working. Then you
+can extend it.
```

Comment:
Is it possible to override the original function by calling it in the new function, without copy-pasting the whole function?

Reply:
No, you attach a whole new function at the place of the old one. The old one is lost.

You can use Inheritance for that, but that means you have to rewrite a constructor (some lines) and replace the "normal" controller with your new "children" controller in ngeo (one line).
So.. well... it's not a bad idea. But I've not tested.

Should I add a sentence that says it's possible ?

Reply:
Oh, and you can't do that if you want to interfere between lines of the functions to override.

Reply:
OK, no forget it, thanks